sys/include/ztimer.h

```diff
@@ -839,6 +839,27 @@ static inline void ztimer_init_extend(ztimer_clock_t *clock)
 }
 #endif /* MODULE_ZTIMER_EXTEND */
 
+/**
+ * @brief   Internal function to sleep on a higher frequency clock for a
+ *          duration specified as lower frequency ticks without overflowing
+ * @param[in,out]   clock       Higher frequency clock to sleep on
+ * @param[in]       time        Time to sleep in lower frequency clock ticks
+ * @param[in]       scale       Each tick in @p time corresponds to @p scale
+ *                              ticks of @p clock
+ */
+static inline void _ztimer_sleep_scale_up(ztimer_clock_t *clock, uint32_t time,
+                                          uint32_t scale)
+{
+    const uint32_t max_sleep = UINT32_MAX / scale;
+
+    while (time > max_sleep) {
+        ztimer_sleep(clock, max_sleep * scale);
```

Comment:
```suggestion
        ztimer_sleep(clock, max_sleep * scale);
```

without `aquire` clock ist allowed to be stop between sleep_calls

Reply:
OK. Other users of this function may actually care about not having the clock powering down and up in between. I added this.